src/Components/Components/src/Routing/Router.cs

```diff
@@ -194,51 +195,58 @@ internal virtual void Refresh(bool isNavigationIntercepted)
 
         private async ValueTask<bool> RunOnNavigateAsync(string path, Task previousOnNavigate)
         {
-            // If this router instance does not provide an OnNavigateAsync parameter
-            // then we render the component associated with the route as per usual.
-            if (!OnNavigateAsync.HasDelegate)
+            if (OnNavigateAsync == null)
             {
                 return true;
             }
```

Comment:
I'm looking at this again for ad-hoc API review and noticed something I missed before. Can OnNavigateAsync ever transition from non-null to null? If so, this should probably be moved below the `await previousOnNavigate;` line, so any previous calls to OnNavigateAsync still get canceled and awaited.

Reply:
Yes, good call. `OnNavigateAsync` can transition from non-null to null if the user changes the parameter in the `Router` component. This will invoke `RunOnNavigateAsync` inside the `SetParametersAsync` call and go through this flow with a null value for `OnNavigateAsync` in which cause it'll return without awaiting the previous task.